Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/View.purs

```diff
@@ -1953,19 +2075,512 @@ rideTrackingView push state =
+              ]
+        ]
+    ]
+
+messageNotificationView :: forall w. (Action -> Effect Unit) -> HomeScreenState -> PrestoDOM ( Effect Unit) w
+messageNotificationView push state =
+  (if state.props.showChatNotification then 
+    if os == "IOS" then PrestoAnim.animationSet[] else PrestoAnim.animationSet [fadeInWithDuration 1000 true]
+  else if state.props.isNotificationExpanded then 
+    if os == "IOS" then PrestoAnim.animationSet[] else PrestoAnim.animationSet [fadeOutWithDuration 1000 true]
+  else  PrestoAnim.animationSet []) $ 
+  linearLayout
+  [ height $ V $ 130
+  , width $ MATCH_PARENT
+  , margin $ MarginBottom 8
+  , padding $ Padding 12 12 12 12
+  , background Color.black900
+  , orientation VERTICAL
+  , clickable true
+  , accessibility $ if state.props.isNotificationExpanded && os /= "IOS" then ENABLE else if not state.props.isNotificationExpanded then DISABLE_DESCENDANT else DISABLE
+  , accessibilityHint $ "Quick Chat : Widget"
+  , onClick push $ const $ MessageDriver
+  , onAnimationEnd push $ const $ MessageViewAnimationEnd
+  , visibility $ if (((any (_ == state.props.currentStage)) [ RideAccepted, ChatWithDriver]) && state.data.currentSearchResultType /= QUOTES && state.data.config.feature.enableChat) && state.data.config.feature.enableSuggestions then VISIBLE else if state.props.currentStage == RideStarted && os == "IOS" then INVISIBLE else GONE
+  , cornerRadius 20.0
+  ][linearLayout 
+    [ height $ WRAP_CONTENT
+    , width $ MATCH_PARENT
+    , clickable true
+    , accessibility DISABLE
+    ][ messagePromtView push state
+     , if ((not $ didDriverMessage FunctionCall) && (state.data.lastSentMessage.sentBy /= "")) then messageView push state.data.lastSentMessage 
+       else if (state.data.lastReceivedMessage.sentBy /= "") then messageView push state.data.lastReceivedMessage 
+       else if os == "IOS" then emptyTextView state else linearLayout[][]
+     , linearLayout
+       [ height $ WRAP_CONTENT
+       , width $ MATCH_PARENT
+       , gravity RIGHT
+       , clickable true
+       , accessibility DISABLE
+       ][ linearLayout
+         [ height $ WRAP_CONTENT
+         , width $ WRAP_CONTENT
+         , cornerRadius 20.0
+         , clickable true
+         , id $ getNewIDWithTag "CrossView"
+         , accessibility ENABLE
+         , accessibilityHint $ "Close : Button : Select to close chat widget"
+         , background Color.manatee200
+         , padding $ Padding 10 10 10 10
+         , onClick push $ const $ RemoveNotification
+         ][imageView
+           [ height $ V 16
+           , width $ V 16
+           , accessibility DISABLE
+           , imageWithFallback $ fetchImage FF_ASSET "ny_ic_cross_white"
+           ]
+         ]
+       ]  
+    ]
+    , separatorView push state
+    , if (state.data.lastMessage.sentBy == "Driver" || not (didDriverMessage FunctionCall)) then quickRepliesView push state else if os == "IOS" then emptyTextView state else linearLayout[][]
```

Comment:
@PraveenGongada why is this check required? 
`if os == "IOS" then emptyTextView state else linearLayout[][]`